pkg/netceptor/ping.go

```diff
@@ -7,8 +7,21 @@ import (
 	"time"
 )
 
-// Ping sends a single test packet and waits for a reply or error.
+// NetcForPing should include all methods of Netceptor needed by the Ping function.
+type NetcForPing interface {
+	ListenPacket(service string) (PacketConner, error)
+	NewAddr(target string, service string) Addr
+	NodeID() string
+	Context() context.Context
+}
+
+// Ping calls CreatePing to sends a single test packet and waits for a reply or error.
 func (s *Netceptor) Ping(ctx context.Context, target string, hopsToLive byte) (time.Duration, string, error) {
+	return CreatePing(ctx, s, target, hopsToLive)
+}
+
+// CreatePing creates Ping by sending a single test packet and waits for a replay or error.
+func CreatePing(ctx context.Context, s NetcForPing, target string, hopsToLive byte) (time.Duration, string, error) {
```

Comment:
`CreatePing` suggests to me that a `ping` struct would be created. However this function seems to send a ping request, I suggest changing the name to something like `SendPing` or `DoPing` or trying to turn it into a verb of some kind